$OpenBSD: patch-src_calibre_constants_py,v 1.4 2016/01/20 23:08:21 sthen Exp $

- add OpenBSD to list of recognized BSDs.

USB (and probably MTP) support needs more work, the current code only
supports OSX, Windows, and Linux SYSFS.

--- src/calibre/constants.py.orig	Fri Jan 15 02:48:37 2016
+++ src/calibre/constants.py	Wed Jan 20 22:22:01 2016
@@ -21,7 +21,8 @@ isnewosx  = isosx and getattr(sys, 'new_app_bundle', F
 isfreebsd = 'freebsd' in _plat
 isnetbsd = 'netbsd' in _plat
 isdragonflybsd = 'dragonfly' in _plat
-isbsd = isfreebsd or isnetbsd or isdragonflybsd
+isopenbsd = 'openbsd' in _plat
+isbsd = isfreebsd or isnetbsd or isdragonflybsd or isopenbsd
 islinux   = not(iswindows or isosx or isbsd)
 isfrozen  = hasattr(sys, 'frozen')
 isunix = isosx or islinux
@@ -158,7 +159,7 @@ class Plugins(collections.Mapping):
             plugins.extend(['winutil', 'wpd', 'winfonts'])
         if isosx:
             plugins.append('usbobserver')
-        if islinux or isosx:
+        if islinux or isosx or isbsd:
             plugins.append('libusb')
             plugins.append('libmtp')
         self.plugins = frozenset(plugins)
